src/osip-monitoring-controller/src/main/java/edu/kit/pse/osip/monitoring/controller/MonitoringController.java

```diff
@@ -80,14 +121,442 @@ private void setupUI(Stage stage) {
+        mixCont.client.connectClient();
+        mixCont.absClient = mixCont.client;
+        
+        for (TankContainer cont : tanks) {
+            cont.client = new TankClient(new RemoteMachine(currentSettings.getHostname(cont.tank.getTankSelector(),
+                    DEFAULT_HOSTNAME), currentSettings.getPort(cont.tank.getTankSelector(), defaultPort++)));
+            cont.client.connectClient();
+            cont.absClient = cont.client;
+        }
     }
+    
+    private void reSetupClients() {
+        int defaultPort = OSIPConstants.DEFAULT_PORT_MIX;
+
+        MixTankClient oldMix = mixCont.client;
+
+        try {
+            mixCont.client = new MixTankClient(
+                    new RemoteMachine(currentSettings.getHostname(mixCont.tank.getTankSelector(),
+                    DEFAULT_HOSTNAME), currentSettings.getPort(mixCont.tank.getTankSelector(), defaultPort++)));
+            mixCont.client.connectClient();
+            mixCont.absClient = mixCont.client;
+            try {
+                oldMix.disconnectClient();
+            } catch (UAClientException e) {
+                System.err.println(e.getMessage());
+            }
+        } catch (UAClientException e) {
+            System.err.println(e.getMessage());
+            mixCont.client = oldMix;
+            mixCont.absClient = mixCont.client;
+        }
+            
+        for (TankContainer cont: tanks) {
+            TankClient old = cont.client;
+            try {
+                cont.client = new TankClient(new RemoteMachine(currentSettings.getHostname(cont.tank.getTankSelector(),
+                        DEFAULT_HOSTNAME), currentSettings.getPort(cont.tank.getTankSelector(), defaultPort++)));
+                cont.client.connectClient();
+                cont.absClient = cont.client;
+                try {
+                    old.disconnectClient();
+                } catch (UAClientException e) {
+                    System.err.println(e.getMessage());
+                }
+            } catch (UAClientException e) {
+                System.err.println(e.getMessage());
+                cont.client = old;
+                cont.absClient = cont.client;
+            }
+        }
+    }
+    
+    private void handleSaveButton(ClientSettingsWrapper settings,
+            MonitoringViewInterface currentMonitoringViewInterface,
+            SettingsViewInterface currentSettingsViewInterface) {
+        for (TankSelector tank: TankSelector.values()) {
+            currentSettings.setFetchInterval(currentSettingsViewInterface.getUpdateInterval());
+            currentSettings.setFillLevelDiagram(tank, currentSettingsViewInterface.isFillLevelProgressionEnabled(tank));
+            currentSettings.setOverflowAlarm(tank, currentSettingsViewInterface.isOverflowAlarmEnabled(tank));
+            currentSettings.setOverheatingAlarm(
+                    tank, currentSettingsViewInterface.isTemperatureOverheatingAlarmEnabled(tank));
```

Comment:
Passt das in eine Zeile, wenn du `currentSettingsViewInterface` ein bisschen kürzer benennst?